### PR TITLE
Improve site a11y

### PR DESCRIPTION
**Pull request by fsansalvadore**
## What kind of change does this PR introduce?

A11y and keyboard navigation UX and styling:

- Improve focus on tab on main pages / components
- Remove double focus on Buttons inside Links
- More semantic html tags

Fixes:
- #15359 

https://github.com/supabase/supabase/assets/25671831/aa74b6f1-5bee-4193-b4a9-002c0d4b640d

https://github.com/supabase/supabase/assets/25671831/744cba09-e7c2-4d0d-96c0-bd509a9ab682

### CODE REVIEW

File: apps/www/components/BuiltWithSupabase/index.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ const BuiltExamples = () => {
         <p className="p">There are many example apps and starter projects to get you started.</p>
         <div className="flex justify-center gap-2 py-4">
           <Link href="/docs/guides/examples" as="/docs/guides/examples" passHref>
-            <a>
+            <a tabIndex={-1}>
```
Comment (MildTomato) [inline apps/www/components/BuiltWithSupabase/index.tsx:19]: you can do 

```
<Link pashref>
  <Button aschild>
    <a>

**Review by MildTomato — CHANGES_REQUESTED** [review]

File: apps/www/styles/index.css [hunk 2]
```diff
@@ -8,6 +8,33 @@
 @import './../../../packages/ui/build/css/themes/dark.css';
 @import './../../../packages/ui/build/css/themes/light.css';
 
+@layer components {
+  .focus-base {
```
Comment (MildTomato) [inline apps/www/styles/index.css:12]: Could put these into the tailwind config maybe.
Don't have to do it in this PR

**Review by MildTomato — COMMENTED** [review]
.

File: apps/www/components/BuiltWithSupabase/index.tsx (hunk 1 above)
Comment (fsansalvadore) [reply apps/www/components/BuiltWithSupabase/index.tsx:19]: Oh, that's how it's composed... I was trying the Button outside of the Link tag and it wasn't working 😁
Will update thanks 👍

File: apps/www/styles/index.css (hunk 2 above)
Comment (fsansalvadore) [reply apps/www/styles/index.css:12]: That would be ideal, but I couldn't figure out a way to use raw tailwind classes to create a new class 🤔

Like:

```js
extend: {
  focus: {
    base: '@apply focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-scale-900 ring-0 transition-all focus-visible:ring-offset-transparent',
  }
}
```

Any idea?

Comment (MildTomato) [reply apps/www/styles/index.css:12]: Need to use a utility for a composition of tailwind.

Could use matchUtilities
https://tailwindcss.com/docs/plugins#dynamic-utilities

Or maybe make a utility just to reset all the focus stuff, so all you need then is something like `focus-reset focus-visible:ring-scale-900`

File: apps/www/components/ExampleCard.tsx [hunk 3]
```diff
@@ -47,7 +47,7 @@ function ExampleCard(props: any) {
         >
           <Link href={props.repo_url} as={props.repo_url} passHref>
             <a
-              className="text-scale-1100 hover:text-scale-1200 flex flex-row items-center text-sm"
+              className="text-scale-1100 hover:text-scale-1200 flex flex-row items-center text-sm focus-base focus-rounded-sm"
```
Comment (MildTomato) [inline apps/www/components/ExampleCard.tsx:50]: Can we remove all the scale stuff as well?

`text`, `text-light` etc

**Review by MildTomato — CHANGES_REQUESTED** [review]
Maybe we are regressing on the tailwind classes

**Pull request closed without merging**